testbench/common.py

```diff
@@ -724,12 +725,22 @@ def handle_retry_test_instruction(database, request, method):
         items = list(retry_connection_matches.groups())
         if items[0] == "reset-connection":
             database.dequeue_next_instruction(test_id, method)
-            fd = request.environ.get("gunicorn.socket", None)
-            if fd is not None:
-                fd.setsockopt(
-                    socket.SOL_SOCKET, socket.SO_LINGER, struct.pack("ii", 1, 0)
-                )
-                fd.close()
+            fd = None
+            channel = None
+            if running_on_windows:
```

Comment:
But this is not about whether you are running on Windows, is it?  It is about whether you are using `waitress`.  For example, if we decided to use `waitress` on Linux this test would be all wrong?  Please fix to test for the presence of `waitress`.

It seems that `request.environ.get("SERVER_SOFTWARE")` can be used for this purpose.